Tests/BindTests/OutputTests.swift

```diff
@@ -0,0 +1,333 @@
+
+    func testCombineArray() {
+        let output1 = Output<Bool>()
+        let output2 = Output<Bool>()
+        let output3 = Output<Bool>()
+
+        var outputValue1: Bool?
+        var outputValue2: Bool?
+        var outputValue3: Bool?
+
+        Output<Bool>.combine(outputs: [output1, output2, output3]).bind { valuesArray in
+            outputValue1 = valuesArray[0]
+            outputValue2 = valuesArray[1]
+            outputValue3 = valuesArray[2]
+        }
+
+        XCTAssertEqual(outputValue1, nil)
+        XCTAssertEqual(outputValue2, nil)
+        XCTAssertEqual(outputValue3, nil)
+
+        output1.update(withValue: true)
+
+        XCTAssertEqual(outputValue1, nil)
+        XCTAssertEqual(outputValue2, nil)
+        XCTAssertEqual(outputValue3, nil)
+
+        output2.update(withValue: true)
+
+        XCTAssertEqual(outputValue1, nil)
+        XCTAssertEqual(outputValue2, nil)
+        XCTAssertEqual(outputValue3, nil)
+
+        output3.update(withValue: true)
+
+        XCTAssertEqual(outputValue1, true)
+        XCTAssertEqual(outputValue2, true)
+        XCTAssertEqual(outputValue3, true)
+
+        output1.update(withValue: false)
+
+        XCTAssertEqual(outputValue1, false)
+        XCTAssertEqual(outputValue2, true)
+        XCTAssertEqual(outputValue3, true)
+    }
+
+    func testCombineTwoTypes() {
+        let output1 = Output<Bool>()
+        let output2 = Output<String>()
+
+        var outputValue1: Bool?
+        var outputValue2: String?
+
+        Output<Bool>
+            .combine(output1, output2)
+            .bind { value1, value2 in
+                outputValue1 = value1
+                outputValue2 = value2
+            }
+
+        XCTAssertNil(outputValue1)
+        XCTAssertNil(outputValue2)
+
+        output1.update(withValue: true)
+
+        XCTAssertNil(outputValue1)
+        XCTAssertNil(outputValue2)
+
+        output2.update(withValue: "test")
+
+        XCTAssertEqual(outputValue1, true)
+        XCTAssertEqual(outputValue2, "test")
+
+        output1.update(withValue: false)
+
+        XCTAssertEqual(outputValue1, false)
+        XCTAssertEqual(outputValue2, "test")
+    }
+
+    func testMap() {
+        //swiftlint:disable:next nesting
+        enum TestEnum {
+            case one
+            case two
+        }
+
+        let value = Output<TestEnum>()
+
+        let mappedValue: Output<String> =
+            value
+                .map { type in
+                    switch type {
+                    case .one:
+                        return "one"
+                    case .two:
+                        return "two"
+                    }
+                }
+
+        value.update(withValue: .one)
+        XCTAssertEqual(mappedValue.latest, "one")
+
+        value.update(withValue: .two)
+        XCTAssertEqual(mappedValue.latest, "two")
+
+        value.update(withValue: .one)
+        XCTAssertEqual(mappedValue.latest, "one")
+                }
+
+    func testFlatMap() {
```

Comment:
this looks like it is the same test as for map? Is it missing testing some of the functionality